reactive/flume.py

```diff
@@ -74,19 +56,29 @@ def waiting_for_flume_available(flume):
 @when('flumeagent.installed', 'flume-agent.available')
 @when_not('flumeagent.started')
 def configure_flume(flumehdfs):
-    from charms.flume import Flume  # in lib/charms; not available until after bootstrap
-
-    port = flumehdfs.get_flume_port()
-    ip = flumehdfs.get_flume_ip()
-    protocol = flumehdfs.get_flume_protocol()
-    flumehdfsinfo = {'port': port, 'private-address': ip, 'protocol': protocol}
-    hookenv.log("Connecting to Flume HDFS at {}:{} using {}".format(port, ip, protocol))
-    hookenv.status_set('maintenance', 'Setting up Flume')
-    flume = Flume(dist_config())
-    flume.configure_flume(flumehdfsinfo)
-    flume.restart()
-    hookenv.status_set('active', 'Ready')
-    set_state('flumeagent.started')
+    try:
+        port = flumehdfs.get_flume_port()
+        ip = flumehdfs.get_flume_ip()
+        protocol = flumehdfs.get_flume_protocol()
+        flumehdfsinfo = {'port': port, 'private-address': ip, 'protocol': protocol}
+        hookenv.log("Connecting to Flume HDFS at {}:{} using {}".format(port, ip, protocol))
+        hookenv.status_set('maintenance', 'Setting up Flume')
+        flume = Flume(dist_config())
+        flume.configure_flume(flumehdfsinfo)
+        flume.restart()
+        hookenv.status_set('active', 'Ready')
+        set_state('flumeagent.started')
+    except:
```

Comment:
Let's rework this try/except.  We should be more explicit about the exception we're expecting to catch (KeyError, whatever).  We don't want this bare exception because it could hide a trace from some other error.